src/dotnet/Api/Identifiers/TranslationSourceId.cs

```diff
@@ -0,0 +1,150 @@
+using System.ComponentModel;
+using ActualChat.Internal;
+using ActualLab.Fusion.Blazor;
+using MemoryPack;
+using MessagePack;
+
+namespace ActualChat;
+
+#pragma warning disable CS0659, CS0660, CS0661 // Type overrides Object.Equals(object o) but does not override Object.GetHashCode()
+#pragma warning disable MA0097 // IComparable should implement <, >, etc.
+
+[DataContract, MemoryPackable(GenerateType.NoGenerate)]
+[JsonConverter(typeof(StringIdentifierJsonConverter<TranslationSourceId>))]
+[Newtonsoft.Json.JsonConverter(typeof(StringIdentifierNewtonsoftJsonConverter<TranslationSourceId>))]
+[MessagePackFormatter(typeof(StringIdentifierMessagePackFormatter<TranslationSourceId>))]
+[TypeConverter(typeof(StringIdentifierTypeConverter<TranslationSourceId>))]
+[ParameterComparer(typeof(ByValueParameterComparer))]
+
+public partial class TranslationSourceId  : StringIdentifier, IStringIdentifier<TranslationSourceId>
+{
+    private static readonly ILruCache<string, TranslationSourceId> Cache = CreateCache<TranslationSourceId>(256);
+
+    public const char Delimiter = ':';
+
+    public static TranslationSourceId New(TextEntryId textEntryId)
+        => new TranslationSourceId(textEntryId.Value, textEntryId);
+
+    public static TranslationSourceId New(ChatId chatId, TranslationIdKind kind, long refLid)
```

Comment:
Make serveral overloads that accepts Conversation and Thread

Reply:
Done